scripts/python_nbd_client.py

```diff
@@ -487,8 +488,9 @@ def connect(self, exportname):
         LOGGER.debug("NBD got size=%d transmission flags=%d",
                      self._size, self._transmission_flags)
         # ignore the zeroes
-        zeroes = self._recvall(124)
-        LOGGER.debug("NBD got zeroes: %s", zeroes)
+        if (self._transmission_flags & NBD_FLAG_NO_ZEROES) == 0:
```

Comment:
NBD_FLAG_NO_ZEROES is not in the transmission flags but in the handshake flags, and if we want to skip zeroes we need to send the same flag to the server: https://github.com/NetworkBlockDevice/nbd/blob/master/doc/proto.md#handshake-flags
So the code is correct because we never send `NBD_FLAG_C_NO_ZEROES` to the server, which means it will never skip the zeroes.

Reply:
Good catch! We should obey the NBD protocol.

I can fix this up on the server side rather than here.